x-pack/plugins/runtime_fields/public/components/runtime_field_form/runtime_field_form.tsx

```diff
@@ -0,0 +1,148 @@
+
+  useEffect(() => {
+    if (onChange) {
+      onChange({ isValid: isFormValid, isSubmitted, submit });
+    }
+  }, [onChange, isFormValid, isSubmitted, submit]);
+
+  return (
+    <Form form={form}>
+      <EuiFlexGroup>
+        {/* Name */}
+        <EuiFlexItem>
+          <UseField path="name" component={TextField} data-test-subj="nameField" />
+        </EuiFlexItem>
+
+        {/* Return type */}
+        <EuiFlexItem>
+          <UseField<EuiComboBoxOptionOption[]> path="type">
+            {({ label, value, setValue }) => {
+              if (value === undefined) {
+                return null;
+              }
+              return (
+                <>
+                  <EuiFormRow label={label} fullWidth>
+                    <EuiComboBox
+                      placeholder={i18n.translate(
+                        'xpack.runtimeFields.form.runtimeType.placeholderLabel',
+                        {
+                          defaultMessage: 'Select a type',
+                        }
+                      )}
+                      singleSelection={{ asPlainText: true }}
+                      options={RUNTIME_FIELD_OPTIONS}
+                      selectedOptions={value}
+                      onChange={(newValue) => {
+                        if (newValue.length === 0) {
+                          // Don't allow clearing the type. One must always be selected
+                          return;
+                        }
+                        setValue(newValue);
+                      }}
+                      isClearable={false}
+                      data-test-subj="typeField"
+                      fullWidth
+                    />
+                  </EuiFormRow>
+                </>
+              );
+            }}
+          </UseField>
+        </EuiFlexItem>
+      </EuiFlexGroup>
+
+      <EuiSpacer size="l" />
+
+      {/* Script */}
+      <UseField<string> path="script">
+        {({ value, setValue, label, helpText, isValid, getErrorsMessages }) => {
+          return (
+            <EuiFormRow
+              label={label}
+              error={getErrorsMessages()}
+              isInvalid={!isValid}
+              helpText={
+                <EuiFlexGroup justifyContent="flexEnd">
+                  <EuiFlexItem grow={false}>
+                    <EuiLink
+                      href={`${docsBaseUri}/to-be-defined`}
+                      target="_blank"
+                      external
+                      data-test-subj="painlessSyntaxLearnMoreLink"
+                    >
+                      {i18n.translate('xpack.runtimeFields.form.script.learnMoreLinkText', {
+                        defaultMessage: 'Learn more about syntax.',
+                      })}
+                    </EuiLink>
+                  </EuiFlexItem>
+                </EuiFlexGroup>
+              }
+              fullWidth
+            >
+              <CodeEditor
+                languageId="painless"
```

Comment:
You should be able to import the language id from kbn-monaco now.

```
import { PainlessLang } from '@kbn-monaco';

<CodeEditor 
  languageId={PainlessLang.ID}
  ....
/>
```

Reply:
Good to know, I will update it. Although IMO it should be the `CodeEditor` that has the `languageId` correctly typed with available options (`'painless'|'other'`)